src/assets.ts

```diff
@@ -663,32 +701,42 @@ export async function addAssetsIfNecessary(server: OmniSharpServer): Promise<Add
     });
 }
 
-async function doesAnyAssetExist(generator: AssetGenerator) {
-    return new Promise<boolean>((resolve, reject) => {
-        fs.exists(generator.launchJsonPath, exists => {
-            if (exists) {
-                resolve(true);
-            }
-            else {
-                fs.exists(generator.tasksJsonPath, exists => {
-                    resolve(exists);
-                });
-            }
-        });
-    });
-}
+async function getExistingAssets(generator: AssetGenerator) {
+    return new Promise<string[]>((resolve, reject) => {
+        let assets: string[] = [];
+        if (fs.pathExistsSync(generator.tasksJsonPath)) {
+            const content = fs.readFileSync(generator.tasksJsonPath).toString();
+            let taskLabels = [ "build", "publish", "watch" ];
+            const tasks = jsonc.parse(content)?.tasks?.
+                map((t: { label: string; }) => t.label).
+                filter((l: string) => taskLabels.includes(l));
+
+            assets = assets.concat(tasks);
+        }
 
-async function deleteAssets(generator: AssetGenerator) {
-    return Promise.all([
-        util.deleteIfExists(generator.launchJsonPath),
-        util.deleteIfExists(generator.tasksJsonPath)
-    ]);
+        if(fs.pathExistsSync(generator.launchJsonPath)) { 
+            const content = fs.readFileSync(generator.launchJsonPath).toString();
+            let configurationNames = [
```

Comment:
Not very happy about this list. I tried sharing the names in constants but it was not much better.